CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+Contributing to Uuid
+---
+[Contributing to Uuid]: #contributing-to-uuid
```

Comment:
Do we want to refer to the crate in prose as _Uuid_, _uuid_ or `uuid`?

Reply:
Hmm... yea wrote this over mutliple days... so inconsistencies exist. How about when referring to the project we use Uuid and `uuid` when its the crate?